### DISCUSSION TITLE
Add Vacation State to Water Heaters

### DISCUSSION THREAD

**Discussion by tpaulus** [home-assistant/architecture#589] [Entity Models] [answered]
Raising as requested by @frenck in https://github.com/home-assistant/core/pull/52494.

## Context
Econet Water Heaters support a Vacation State which is not modeled in the integration today, nor are supported by the Water Heater Entity.

## Proposal
* Add Vacation State to list of Operating Modes to allow a mapping of all possible econet states to HA states.
This work may need to be combined with additional changes as being discussed in https://github.com/home-assistant/architecture/discussions/518.

**Comment by frenck** [comment]
So, Home Assistant tries to be a generic interface covering most common use cases, we currently support 10 integrations with water heaters; possibly more via custom integration.

Do others need this? As in, is this a common thing?

**Reply by tpaulus** [reply]
I’m not sure, but as it stands, there isn’t a way to set the operating mode to “vacation” as it isn’t modeled in HA, which is the issue I am trying to address. Additional, HA cannot render the state of the water heater if it is in vacation mode as the mapping from the vendor state to HA state is missing the Vacation Mode.

**Reply by frenck** [reply]
> I’m not sure, but as it stands, there isn’t a way to set the operating mode to “vacation” as it isn’t modeled in HA, which is the issue I am trying to address.

Yes; however, we don't want to add everything to our model; unless it has a broader use case, hence my question.

**Reply by iMicknl** [reply]
Would it make sense to add presets to the water_heater? In Somfy TaHoma (custom component) we face the issue that we don't have presets. https://github.com/iMicknl/ha-tahoma/issues/337

To be honest, the water heater entity could use some more love, but I haven't had time lately to have a better look. For reference, see this discussion https://github.com/home-assistant/architecture/discussions/518 around the water heater entity with a summary of all water heaters and their features.

**Comment by make-all** [comment]
A model "Thermex series V pro" requested for my custom integration here https://github.com/make-all/tuya-local/pull/769 would benefit from this.

It has modes "Economy", "Optimal", "Turbo" and "No Frost" based on the original mobile app (possibly translated, as Anti-frost would be more common in English).  I'm thinking of eco, electric, performance as the mappings for the first three, but there is no obvious standard mapping for the "No Frost".

Personally I think the mode should be "away" to be consistent with the climate platform's standard presets.

**Reply by make-all** [reply]
It seems water_heater does already support away mode, but as a separate service call, not as an operation mode.  This is awkward to handle for those heaters that implement it as just another operation mode, but the reverse will be true for the devices that have a separate switch for it, which must be the case for whatever device triggered this implementation decision.  Since a solution that fits all devices perfectly is not possible, I think working with the current architecture is probably the best, as duplicating the same functionality in different way would be confusing not just for developers, but for end users too.

**Comment by frenck** [comment, marked as answer]
We have discussed this proposal, and for now decided that doesn't get enough convincing reasoning to accept this in this moment in time. We've decided for now not to accept the proposal.

Maybe something we can reconsider in the future.

../Frenck
